src/podutils.rs

```diff
@@ -93,46 +96,229 @@ where
     )
 }
 
-/// Checks whether the given pod is assigned to (via the `spec.node_name` field) the given `node_name`.
-pub fn is_pod_assigned_to_node(pod: &Pod, node_name: &str) -> bool {
+/// Checks whether the given Pod is assigned to (via the `spec.node_name` field) the given `node_name`.
+pub fn is_pod_assigned_to_node_name(pod: &Pod, node_name: &str) -> bool {
     matches!(pod.spec, Some(PodSpec { node_name: Some(ref pod_node_name), ..}, ..) if pod_node_name == node_name)
 }
 
+/// Checks whether the given Pod is assigned to (via the `spec.node_name` field) the given Node (via `metadata.name`).
+pub fn is_pod_assigned_to_node(pod: &Pod, node: &Node) -> bool {
+    matches!((pod.spec.as_ref(), node.metadata.name.as_ref()),
+        (
+            Some(PodSpec { node_name: Some(ref pod_node_name), ..}, ..),
+            Some(node_node_name),
+        ) if pod_node_name == node_node_name
+    )
+}
+
+/// This method checks if a Pod contains all required labels including an optional value per label.
+///
+/// # Arguments
+///
+/// - `pod` - the Pod to check for labels
+/// - `required_labels` - is a BTreeMap of label keys to an optional value
+///
+/// # Example
+///
+/// ```
+/// use stackable_operator::podutils;
+/// # use k8s_openapi::api::core::v1::{Pod, PodSpec};
+/// # use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+/// use std::collections::BTreeMap;
+///
+/// # let pod = Pod {
+/// #            metadata: ObjectMeta {
+/// #                ..ObjectMeta::default()
+/// #            },
+/// #            spec: None,
+/// #            status: None,
+/// #        };
+///
+/// let mut required_labels = BTreeMap::new();
+/// required_labels.insert("foo".to_string(), Some("bar".to_string()));
+///
+/// assert!(!podutils::pod_matches_labels(&pod, &required_labels));
+/// ```
+pub fn pod_matches_labels(pod: &Pod, required_labels: &BTreeMap<String, Option<String>>) -> bool {
```

Comment:
I've commented somewhere else if we could introduce a type alias for `BTreeMap<String, Option<String>>` - after seeing this, we could think that a bit further and make that `BTreeMap<String, Option<Vec<String>>>` instead and implement `From<BTreeMap<String, Option<String>>` for that, which would allow us to remove this function and have `pod_matches_multiple_label_values` take something implementing Into<LabelList> or whatever we decide to call it.

Just a thought, and doesn't need to be addressed for this PR though.

Reply:
Do you have a suggestion for a type alias?
I thought about it as well/had one in for a while but nothing really descriptive came to mind.

Reply:
I've spent 10 minutes with my idea, which of course wasn't all that easy, because you cannot implement traits for an alias, so it had to be a new type..

I ended up with this:

```rust
struct LabelAndValueList {
    list: BTreeMap<String, Option<Vec<String>>>,
}

impl From<&BTreeMap<String, Option<Vec<String>>>> for LabelAndValueList {
    fn from(value: &BTreeMap<String, Option<Vec<String>>>) -> Self {
        LabelAndValueList {
            list: value.clone(),
        }
    }
}

impl From<&BTreeMap<String, Option<String>>> for LabelAndValueList {
    fn from(value: &BTreeMap<String, Option<String>>) -> Self {
        let result = value
            .iter()
            .map(|(key, value)| {
                (
                    key.clone(),
                    value.as_ref().map(|string| vec![string.clone()]),
                )
            })
            .collect::<BTreeMap<_, _>>();
        LabelAndValueList { list: result }
    }
}
```

Which would allow having only one function that can take both types of BTreeMaps: 

```rust
pub fn pod_matches_multiple_label_values<T>(pod: &Pod, required_labels: T) -> bool
where
    T: Into<LabelAndValueList> + Sized,
{
    let required_labels = required_labels.into();
    // ...
}
```

But probably has other issues. I only tested this until I got one test running :)

Reply:
Interesting. I have to think about it.


I remember  a problem that one list should not be used in place of the other because of different semantic meaning.

I have to admit that I forgot parts of it but one example:
Let's say you have a bunch of role-groups A, B and C and now you find a group "D" in your labels.
This is probably not illegal. More likely is that "D" did exist in your spec and the user removed it.

I'm not sure if this is true here. These things are not easy :)

Reply:
After discussing this offline we decided to not do this because it might mean that someone passes in multiple label values where only one is allowed.
By having two separate methods the distinction becomes clearer.